pages/blog/graphql-codegen-and-apollo-client-3.mdx

```diff
@@ -0,0 +1,176 @@
+import withArticle from '../../ui/blog/article';
+
+export const meta = {
+  title: 'Improve your Apollo-Client v3 type-safety with GraphQL Codegen',
+  author: 'dotan',
+  tags: ['graphql', 'codegen', 'typescript', 'react', 'angular', 'apollo'],
+  date: '2020-10-26',
+  description: 'Get the most out of your type-system using GraphQL Codegen',
+  image: '/blog-assets/graphql-code-generator/cover.png',
+};
+
+export default withArticle({ ...meta });
+
+We've recently introduced improvements for GraphQL-Codegen that can quickly improve your workflow and type-safety with Apollo-Client v3.
+
+I'll try to cover the popular codegen plugins you can use, that already works with Apollo-Client v3.
+
+# possibleTypes
+
+If you are already familiar with GraphQL-Codegen and the plugins it offers, you probably know `fragment-matcher` plugin.
+
+In Apollo-Client v3, [the structure for fragment matcher has been changed, and now it's called `possibleTypes`](https://www.apollographql.com/docs/react/data/fragments/#using-fragments-with-unions-and-interfaces).
+
+The `@graphql-codegen/fragment-matcher@2.0.0` now supports Apollo-Client v3 by default, and it generates type signature and the `possibleTypes` object automatically based on your GraphQL schema.
+
+Here's an example of using it with a codegen.yml file:
+
+```yaml
+schema: my-schema.graphql
+generates:
+  ./src/possible-types.ts:
+    plugins:
+      - fragment-matcher
+```
+
+Then, when you create your Apollo Client cache instance, use the generated variable:
```

Comment:
maybe you can mention how you would do it if you had to do it manually and what are the benefits?